library/kani/src/mem_init.rs

```diff
@@ -115,6 +115,23 @@ impl MemoryInitializationState {
                 self.tracked_offset += to_offset - from_offset;
                 // Note that this preserves the value.
             }
+        } else if self.tracked_object_id == to_obj
+            && self.tracked_offset >= to_offset
+            && self.tracked_offset < to_offset + num_elts * LAYOUT_SIZE
+        {
+            self.value = true;
+        }
```

Comment:
If we are copying to an object that is currently tracked, need to unconditionally initialize the value, since the actual `copy` execution check will happen in a different non-deterministic branch.